eppo/src/main/java/cloud/eppo/android/ConfigurationRequestor.java

```diff
@@ -7,27 +7,34 @@
 import com.google.gson.JsonIOException;
 import com.google.gson.JsonSyntaxException;
 
-import java.io.Reader;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
 
 import cloud.eppo.android.dto.FlagConfig;
 
 public class ConfigurationRequestor {
     private static final String TAG = logTag(ConfigurationRequestor.class);
 
-    private EppoHttpClient client;
-    private ConfigurationStore configurationStore;
+    private final EppoHttpClient client;
+    private final ConfigurationStore configurationStore;
 
     public ConfigurationRequestor(ConfigurationStore configurationStore, EppoHttpClient client) {
         this.configurationStore = configurationStore;
         this.client = client;
     }
 
     public void load(InitializationCallback callback) {
+        // We have two at-bats to load the configuration: loading from cache and fetching
+        // The below variables help them keep track of each other's progress
+        AtomicBoolean cacheLoadInProgress = new AtomicBoolean(true);
+        AtomicReference<String> fetchErrorMessage = new AtomicReference<>(null);
```

Comment:
Can you see this behavior being configurable based on if the user wants to wait for the network request to finish?

Reply:
Yes! I think everything should confirm to [this plan](https://www.notion.so/eppo/Client-SDK-Caching-and-Fetching-Behavior-a9c8caf98c7e4a27b3d4c7751ad65cd9?pvs=4#4d1575abf9e64f0cbd25cf68df470d0d).

For now, this is mirroring how it behaved before (minus the shared preferences, which got ripped out).